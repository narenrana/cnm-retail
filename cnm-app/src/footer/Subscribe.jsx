import React from "react";
import Typography from "@material-ui/core/Typography";
import InputBase from "@material-ui/core/InputBase";
import Fab from "@material-ui/core/Fab";
import useStyles from "./style";
import { Grid } from "@material-ui/core";
import useMediaQuery from "@material-ui/core/useMediaQuery";

function SubscribeGrid() {
  const classes = useStyles();
  return (
    <div>
      <Grid container spacing={0}>
        <Grid item xs={12} md={12} xl={12}>
          <Typography className={classes.subscribeTypehead}>
            Subscribe to our news latter and receive exclusive offer every week
          </Typography>
        </Grid>
        <Grid item xs={12} md={8} xl={8}>
          <InputBase
            className={classes.subscribeInput}
            placeholder="Enter your email"
          />
        </Grid>
        <Grid item xs={12} md={4} xl={4}>
          <Fab
            variant="extended"
            size="small"
            color="primary"
            aria-label="subscribe"
            className={classes.subscribeButton}
          >
            <div className={classes.subscribeButtonText}>Subscribe</div>
          </Fab>
        </Grid>
      </Grid>
    </div>
  );
}

export default function Subscribe() {
  const classes = useStyles();
  const matches = useMediaQuery("(min-width:600px)");

  return matches ? (
    <div className={classes.subscribeRoot}>
      <div className={classes.subscribeMenu}>
        <SubscribeGrid />
      </div>
    </div>
  ) : (
    <div>
      <SubscribeGrid />
    </div>
  );
}
