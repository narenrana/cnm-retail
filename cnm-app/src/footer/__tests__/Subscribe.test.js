/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import Subscribe from "../Subscribe";

describe("Subscribe test", () => {
  it("Subscribe should match snapshot", () => {
    const component = renderer.create(<Subscribe />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
