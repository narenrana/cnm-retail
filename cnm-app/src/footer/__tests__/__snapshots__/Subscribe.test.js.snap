// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscribe test Subscribe should match snapshot 1`] = `
<div>
  <img
    alt="Banner"
    className="makeStyles-footerBanner-2"
    src="/static/images/banner.png"
  />
  <div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-xl-12"
      >
        <p
          className="MuiTypography-root makeStyles-subscribeTypehead-7 MuiTypography-body1"
        >
          Subscribe to our news latter and receive exclusive offer every week
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 MuiGrid-grid-xl-8"
      >
        <div
          className="MuiInputBase-root makeStyles-subscribeInput-8"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter your email"
            type="text"
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-xl-4"
      >
        <button
          aria-label="subscribe"
          className="MuiButtonBase-root MuiFab-root makeStyles-subscribeButton-9 MuiFab-extended MuiFab-sizeSmall MuiFab-primary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiFab-label"
          >
            <div
              className="makeStyles-subscribeButtonText-10"
            >
              Subscribe
            </div>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
